snapcraft/internal/sources/_git.py

```diff
@@ -90,3 +90,25 @@ def pull(self):
             self._pull_existing()
         else:
             self._clone_new()
+        self.assets = self._get_source_details()
+
+    def _get_source_details(self):
```

Comment:
So let's fast-forward here: how do you envision this information being used when we actually save it into the `$SNAP/snap/snapcraft.yaml` and need to re-consume it? For example, just a few lines above there's the logic for erroring out if both `source-commit` and `source-branch` are provided, and yet you're saving both here (among other things). How do you plan on that being disambiguated?